JuniorStart/Factories/ModelFactory.cs

```diff
@@ -0,0 +1,88 @@
+using System.Collections.Generic;
+using System.Linq;
+using JuniorStart.DTO;
+using JuniorStart.Entities;
+
+namespace JuniorStart.Factories
+{
+    public class ModelFactory : IModelFactory
+    {
+        public RecruitmentInformationViewModel Create(RecruitmentInformation recruitmentInformation)
+        {
+            return new RecruitmentInformationViewModel
+            {
+                Id = recruitmentInformation.Id,
+                City = recruitmentInformation.City,
+                Notes = recruitmentInformation.Notes,
+                CompanyName = recruitmentInformation.CompanyName,
+                OwnerId = recruitmentInformation.Owner.Id,
+                WorkPlace = recruitmentInformation.WorkPlace,
+                LinkToApplication = recruitmentInformation.LinkToApplication,
+                CompanyReply = recruitmentInformation.CompanyReply,
+                DateOfCompanyReply = recruitmentInformation.DateOfCompanyReply
+            };
+        }
+
+        public TaskViewModel Create(Task task)
+        {
+            return new TaskViewModel
+            {
+                Id = task.Id,
+                Description = task.Description,
+                Status = task.Status,
+                TodoListId = task.TodoList.Id
+            };
+        }
+
+        public TodoListViewModel Create(TodoList todoList)
+        {
+            List<TaskViewModel> tasks = todoList.Tasks.Select(Create).ToList();
+            return new TodoListViewModel
+            {
+                Id = todoList.Id,
+                Name = todoList.Name,
+                Tasks = tasks
+            };
+        }
+
+        public UserViewModel Create(User user)
+        {
+            return new UserViewModel
+            {
+                Id = user.Id,
+                Email = user.Email,
+                Login = user.Login,
+                FirstName = user.FirstName,
+                LastName = user.LastName
+            };
+        }
+
+        public RecruitmentInformation Map(RecruitmentInformationViewModel recruitmentInformation)
+        {
+            return new RecruitmentInformation
+            {
+                City = recruitmentInformation.City,
+                Notes = recruitmentInformation.Notes,
+                OwnerId = recruitmentInformation.OwnerId,
+                CompanyName = recruitmentInformation.CompanyName,
+                CompanyReply = recruitmentInformation.CompanyReply,
+                WorkPlace = recruitmentInformation.WorkPlace,
+                LinkToApplication = recruitmentInformation.LinkToApplication,
+                DateOfCompanyReply = recruitmentInformation.DateOfCompanyReply
+            };
+        }
+
+        public User Map(UserViewModel userViewModel)
+        {
```

Comment:
nie powinno być mapowania z ViewModelu na normalny obiekt 
+w jakiej sytuacji to miałoby działać? 
od użytkownika nie powinieneś dostawać ViewModelu, tylko osobny request 

osobiście lubie komunikować się z użytkownikiem w kontrolerach za pomocą obiektów Request które on mi przysyła i Response które ja mu wysyłam 
np

public CreateUserResponse CreateUser(CreateUserRequest request)

gdzie jasno mam ustalone co ta metoda w api ma zwrócić użytkownikowi
i jasno określone co użytkownik mi wysyła
troche więcej pracy, ale na dużą skale to świetne rozwiązanie, proste do utrzymania i łatwe do zachowania zasady SRP z SOLID
każdy obiekt ma jedno konkretne zastosowanie, więc nie ma sytuacji gdzie zmiana czegoś powoduje dziwne skutki uboczne 

no z tym że to dodatkowe kilka minut pracy